### PR TITLE
Update local dev setup

### LINKED ISSUE #89: Update local dev setup
When in dev mode:
- run local instance of ganache connected to mainnet as part of the dev server (or in the browser as a service task (should persist when refreshing the page))
- add a button which auto connects metamask to local ganache instance
- add input field for setting the block timestamp / number of the local ganache instance and refresh the browser
- ganache should not be bundled when deploying to staging or prod.

### PR DESCRIPTION
**Pull request by SolcAndMe**
closes #89

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,10 +32,13 @@
     "@types/react": "^18.0.9",
     "@typescript-eslint/eslint-plugin": "^5.41.0",
     "@typescript-eslint/parser": "^5.41.0",
+    "@walletconnect/client": "1.6.2",
```
Comment (SolcAndMe) [inline package.json:35]: I think this can be a dev dependency if it's only used in the impersonate script

File: package.json [hunk 2]
```diff
@@ -29,10 +32,13 @@
...
+    "dotenv": "^16.0.3",
     "eslint": "^8.26.0",
     "eslint-config-next": "^12.1.6",
     "eslint-import-resolver-typescript": "^3.5.2",
     "eslint-plugin-react-hooks": "^4.6.0",
+    "lokijs": "^1.5.12",
```
Comment (SolcAndMe) [inline package.json:41]: Where is the this dependency required?

File: scripts/testnet.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+fi
+
+if [ -z "$NEXT_PUBLIC_ALCHEMY_API_KEY" ]; then
+  echo "NEXT_PUBLIC_ALCHEMY_API_KEY is undefined in .env";
+  exit 1;
+fi
+
+# hardcoded unlocked accounts hold USDC, DAI and ETH necessary for testing on mainnet
+cmd="npx ganache \
+	--fork.url=https://eth-mainnet.alchemyapi.io/v2/$NEXT_PUBLIC_ALCHEMY_API_KEY \
```
Comment (SolcAndMe) [inline scripts/testnet.sh:19]: indentation

File: src/actions.ts [hunk 4]
```diff
@@ -224,7 +226,8 @@ export const getEarnableRate = async (fiat: any, collateralTypesData: any) => {
   const queries = collateralTypesData.flatMap((collateralTypeData: any) => {
     const { properties } = collateralTypeData;
     const { vault, tokenId, vaultType, tokenScale, underlierScale, maturity } = properties;
-    if (new Date() >= new Date(Number(maturity.toString()) * 1000)) return [];
+    const date = USE_FORK ? useStore.getState().ganacheTime.getTime() : new Date();
```
Comment (SolcAndMe) [inline src/actions.ts:229]: move getTimestamp method to utils to reduce duplication

File: src/components/CollateralTypesTable.tsx [hunk 5]
```diff
@@ -19,6 +20,7 @@ export const CollateralTypesTable = () => {
   const fiat = useStore((state) => state.fiat);
   const setSelectedPositionId = useStore((state) => state.setSelectedPositionId);
   const setSelectedCollateralTypeId = useStore((state) => state.setSelectedCollateralTypeId);
+  const ganacheTime = useStore((state) => state.ganacheTime);
```
Comment (SolcAndMe) [inline src/components/CollateralTypesTable.tsx:23]: move to utils

File: src/components/HeaderBar.tsx [hunk 6]
```diff
@@ -16,42 +17,130 @@ interface BlockSyncStatus {
   message: string;
 }
 
+const SESSION_STORAGE_KEY = 'fiat-ui-snapshotIds';
```
Comment (SolcAndMe) [inline src/components/HeaderBar.tsx:20]: Would move forward, revert controls into a separate component

File: src/components/PositionModal/PositionModal.tsx [hunk 7]
```diff
@@ -51,18 +52,28 @@ const PositionModalBody = (props: PositionModalProps) => {
...
   const modifyPositionData = useStore((state) => state.modifyPositionData);
   const selectedCollateralTypeId = useStore((state) => state.selectedCollateralTypeId);
   const selectedPositionId = useStore((state) => state.selectedPositionId);
+  const ganacheTime = useStore((state => state.ganacheTime));
+  const getGanacheTime = useStore((state) => state.getGanacheTime);
 
   const vaultType = modifyPositionData?.collateralType?.properties?.vaultType;
 
   const matured = React.useMemo(() => {
+    if (USE_FORK) {
```
Comment (SolcAndMe) [inline src/components/PositionModal/PositionModal.tsx:65]: move to utils

File: src/components/PositionsTable.tsx [hunk 8]
```diff
@@ -20,8 +21,10 @@ export const PositionsTable = () => {
   const fiat = useStore((state) => state.fiat);
   const setSelectedPositionId = useStore((state) => state.setSelectedPositionId);
   const setSelectedCollateralTypeId = useStore((state) => state.setSelectedCollateralTypeId);
+  const ganacheTime = useStore((state) => state.ganacheTime);
```
Comment (SolcAndMe) [inline src/components/PositionsTable.tsx:24]: move to utils

File: src/state/stores/globalStore.ts [hunk 9]
```diff
@@ -80,6 +84,15 @@ const useStore = create<any>()((set: any) => ({
       modifyPositionData: data
     }))
   },
+  getGanacheTime: async () => {
```
Comment (SolcAndMe) [inline src/state/stores/globalStore.ts:87]: move to utils

File: src/utils.ts [hunk 10]
```diff
@@ -33,9 +35,9 @@ export const commifyToDecimalPlaces = (value: BigNumber, scale: number, decimalP
 };
 
 export const earnableRateToAPY = (earnableRate: BigNumber, maturity: BigNumberish): BigNumber => {
-  const now = Math.floor(Date.now() / 1000)
+  const now = USE_FORK ? Math.floor(useStore.getState().ganacheTime.getTime() / 1000) : Math.floor(Date.now() / 1000);
```
Comment (SolcAndMe) [inline src/utils.ts:38]: encapsulate into separate getTimestamp method

**Comment by SolcAndMe** [comment]
Disable Metamask provider, clear session if `NEXT_PUBLIC_GANACHE_LOCAL` is true

File: package.json (hunk 1 above)
Comment (jFarms) [reply package.json:35]: This is already in dev dependencies :)

File: package.json (hunk 2 above)
Comment (jFarms) [reply package.json]: Walletconnect listed it as a peer dependency for install steps, but it looks like it was only for their "sign client" which we arent using. Removed.

File: scripts/testnet.sh (hunk 3 above)
Comment (jFarms) [reply scripts/testnet.sh]: Fixed, it was not rendering like that in my IDE. Probably hidden tabs/spaces

File: src/actions.ts (hunk 4 above)
Comment (jFarms) [reply src/actions.ts:229]: Fixed

File: src/components/CollateralTypesTable.tsx (hunk 5 above)
Comment (jFarms) [reply src/components/CollateralTypesTable.tsx:23]: fixed

File: src/components/PositionsTable.tsx (hunk 8 above)
Comment (jFarms) [reply src/components/PositionsTable.tsx:24]: fixed

File: src/state/stores/globalStore.ts (hunk 9 above)
Comment (jFarms) [reply src/state/stores/globalStore.ts:87]: I would like to leave this here so we are caching the timestamp.

Comment (jFarms) [reply src/state/stores/globalStore.ts:87]: I would like to keep ganache time in the store to cache the value and to allow collateral and positions data to rerender on updates. 

The tables were not updating to have the new value when they were using only the utils method and not the zustand store hook.

File: src/components/PositionModal/PositionModal.tsx (hunk 7 above)
Comment (jFarms) [reply src/components/PositionModal/PositionModal.tsx]: done

**Pull request merged**